Trigger/src/PrescaleEvent_module.cc

```diff
@@ -25,48 +28,68 @@ namespace mu2e
 
   class PrescaleEvent : public art::EDFilter {
 
-    public:
+  public:
 
-      explicit PrescaleEvent(fhicl::ParameterSet const & p);
+    using  Name    = fhicl::Name;
+    using  Comment = fhicl::Comment;
 
-      // The destructor generated by the compiler is fine for classes
-      // without bare pointers or other resource use.
-      // Plugins should not be copied or assigned.
+    struct EventModeConfig {
+      //fhicl::Atom<uint8_t>        eventMode{ Name("eventMode"), Comment("EventMode: OnSpill, OffSpill, Calib, ...")};//FIXME!!
+      fhicl::Atom<std::string>        eventMode{ Name("eventMode"), Comment("EventMode: OnSpill, OffSpill, Calib, ...")};//FIXME!!
+      fhicl::Atom<float >         prescale { Name("prescale") , Comment("Prescale factor")};
+    };
 
-      PrescaleEvent(PrescaleEvent const &) = delete;
-      PrescaleEvent(PrescaleEvent &&) = delete;
-      PrescaleEvent & operator = (PrescaleEvent const &) = delete;
-      PrescaleEvent & operator = (PrescaleEvent &&) = delete;
+    struct Config {
+      fhicl::Atom<int>                                 debug     { Name("debugLevel")     , Comment("Debug Level"), 0};
+      fhicl::Sequence<fhicl::Table<EventModeConfig>>   eventMode { Name("eventModeConfig"), Comment("List of the Event modes with their PS factor")};
+    };
+    using        Parameters = art::EDFilter::Table<Config>;
+    explicit     PrescaleEvent(const Parameters& conf);
 
-      bool filter(art::Event & e) override;
-      virtual bool endRun(art::Run& run ) override;
+    // The destructor generated by the compiler is fine for classes
+    // without bare pointers or other resource use.
+    // Plugins should not be copied or assigned.
 
-    private:
+    PrescaleEvent(PrescaleEvent const &) = delete;
+    PrescaleEvent(PrescaleEvent &&) = delete;
+    PrescaleEvent & operator = (PrescaleEvent const &) = delete;
+    PrescaleEvent & operator = (PrescaleEvent &&) = delete;
 
-    uint32_t     nPrescale_;
-    bool         useFilteredEvts_;
-    int          _debug;
-    //    TriggerFlag  _trigFlag;
-    unsigned     _nevt, _npass;
+    bool filter(art::Event & e) override;
+    virtual bool endRun(art::Run& run ) override;
+
+  private:
+
+    std::vector<EventModeConfig> _eventMode;
```

Comment:
Why is this a vector?  Can a single filter instance be used in multiple event modes?

Reply:
That's the current idea. The current design is to have the possibility to run the same trigger path in different event-modes with different prescale factors.